tls/s2n_client_hello.c

```diff
@@ -348,17 +348,19 @@ static S2N_RESULT s2n_client_hello_process_cb_response(struct s2n_connection *co
 
 int s2n_client_hello_recv(struct s2n_connection *conn)
 {
-    POSIX_ENSURE(conn->client_hello.callback_async_blocked == 0, S2N_ERR_ASYNC_BLOCKED);
+    if (conn->client_hello_callback_poll == 0) {
+        POSIX_ENSURE(conn->client_hello.callback_async_blocked == 0, S2N_ERR_ASYNC_BLOCKED);
+    }
 
     if (conn->config->client_hello_cb_mode == S2N_CLIENT_HELLO_CB_BLOCKING ||
-            !conn->client_hello.callback_async_done)
+            (!conn->client_hello.callback_async_done && !conn->client_hello.callback_invoked))
     {
         /* Parse client hello */
         POSIX_GUARD(s2n_parse_client_hello(conn));
     }
```

Comment:
I am not 100% sure if this is correct? Can the parse_client_hello span multiple calls?

Reply:
I think this needs to happen only once so what you have should be correct

Reply:
Spoke with @lrstewart and the the `s2n_parse_client_hello` only needs to be called once (infact the function was only being called once before).